src/components/visualizations/utils/color.ts

```diff
@@ -160,7 +161,10 @@ function lighter(color: number, percent: number) {
     return Math.round((t - color) * p) + color;
 }
 
-function formatColor(red: number, green: number, blue: number) {
+function formatColor(red: number, green: number, blue: number, opacity?: number): string {
```

Comment:
`formatColor` should be consistent with `getCalculatedColors` and `getColorPalette`
Please add the default value for `opacity`

Reply:
Updated